src/providers/fail_over.c

```diff
@@ -1188,9 +1189,18 @@ fo_resolve_service_done(struct tevent_req *subreq)
                                     &common->rhostent);
     talloc_zfree(subreq);
     if (ret != EOK) {
-        DEBUG(SSSDBG_CRIT_FAILURE, "Failed to resolve server '%s': %s\n",
+        if (resolv_status == ARES_EFILE) {
+            /* resolv_strerror(resolv_status) provided msg from c-ares lib.
+             * c-ares lib in most distros will default to /etc/hosts for
+             * file based host resolving */
+            DEBUG(SSSDBG_CRIT_FAILURE, "Failed to resolve server '%s': %s "
+                  "[%s]\n", common->name,
```

Comment:
I think this is last comment I will have:)
This is quite odd place to break string into multiple lines, how about something like bellow:
```
DEBUG(SSSDBG_CRIT_FAILURE, "Failed to resolve server '%s': %s [%s]\n",
      common->name,
      resolv_strerror(resolv_status),
      _PATH_HOSTS);
```